spec/models/project_spec.rb

```diff
@@ -136,6 +136,65 @@
     end
   end
 
+  describe "#update_repository" do
```

Comment:
There isn't much logic in the `Project.update_repository` method. Should these specs live in the specs for `PersistRepositoryFromUpstream`?

Reply:
these are specifically about creating the Project.repository_id => Repository association, while that class is about filling in a Repository, I think